### ISSUE TITLE
Staging Review finding: Helper text may be missed by screen reader users

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#103790]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Benefits Dependent Experience Team
**Product:** 21-686c and 21-674
**Feature:** Critical Form Updates

## Findings details
**VA.gov Experience Standard - issue:** [User has enough information to complete a task.](https://design.va.gov/about/experience-standards/#user-has-enough-information-to-complete-a-task)
**VA.gov Experience Standard - category:** [Comprehension](https://design.va.gov/about/experience-standards/#category-comprehension)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

On [/current-spouse-marriage-history/0/location-where-marriage-ended](https://staging.va.gov/view-change-dependents/add-remove-form-21-686c-v2/current-spouse-marriage-history/0/location-where-marriage-ended?add=true), there's helper text "If they got a divorce or annulment..." that appears after the `legend` "Where did the marriage end?"

Even though this helper text appears inside the `fieldset`, it's not programmatically associated with the inputs and won't be announced when a screen reader tabs directly to the first input. I don't think these are super critical instructions, but they're helpful for understanding how to respond.

This also occurs on:
- [/veteran-marriage-history/0/location-where-marriage-ended](https://staging.va.gov/view-change-dependents/add-remove-form-21-686c-v2/veteran-marriage-history/0/location-where-marriage-ended?add=true)

<details><summary>Link, screenshot or steps to recreate</summary>

!["Where did the marriage end?" followed by a paragraph of helper text, then followed by a set of inputs.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/74059-Brian-3.png?raw=true)

</details>

## Recommended action

Move the helper text inside the `legend`. You can still style it to match its current appearance, but putting that text inside the `legend` ensures it's announced.

## References

- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by Midge-dev** [comment]
We have opened a platform ticket for the `fileInputUI` [here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3858). This file upload impostor component is being used instead of `fileInputUI` because it doesn't have a working review page component built in to the pattern. We will continue looking for a solution for this ticket, but this is a platform level problem with the pattern.

cc: @steele-lm @mjknight50

**Comment by Midge-dev** [comment]
@DLarson-Oddball

**Comment by steele-lm** [comment]
@Midge-dev Can you investigate? I think we had previously addressed this issue with [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/103788).

**Comment by Midge-dev** [comment]
It appears that using RJSF's `ui:description` renders this text outside of the legend for some reason. Previously, this was how we did helper text. We may need to build this to be a custom component for now and notify Platform that ui:description isn't rendering inside of a fieldset

**Comment by DLarson-Oddball** [comment]
@Midge-dev Verified this is working in staging 👍
